### DISCUSSION TITLE
Difference between Button and Link

### DISCUSSION THREAD

**Discussion by hasantahir** [godbout/Wooshy.docs#89] [Q&A] [answered]
Hi,

Is there a way to learn which UI element is a `button` and which is a `link`? On this website, Inspector categories similar elements differently:

<img width="525" alt="image" src="https://user-images.githubusercontent.com/11236148/205290498-462af6f2-f0c7-4527-a39c-cb7820920ac9.png">

<img width="477" alt="image" src="https://user-images.githubusercontent.com/11236148/205290521-0c6c4334-c4d5-491a-9cbb-3c8d7796c572.png">

<img width="464" alt="image" src="https://user-images.githubusercontent.com/11236148/205290545-5445635d-474c-432a-afc1-764b0b1fc365.png">

Asking this because would love to quickly reach buttons or links in apps and web pages.

Cheers

**Comment by godbout** [comment, marked as answer]
i can empathize with your pain. the issue is that mostly web pages and Electron apps just do the duck they want. so you have to pray that the devs are not complete morons 😂 when you do a native macOS app, you use nails and bolts provided but Apple. everything comes out of the box and is labeled correctly. when you do a web/Electron app, you have to define everyone yourself. usually, Accessibility  usability is not a priority, or is not that properly and standardly defined. that's why you end up inconsistent data. unfortunately Wooshy doesn't have any decisional power in this (except through reporting to the sites/apps). all it does is grab the data provided by the apps or sites. my "advise" would be to try what _YOU_ think is the role of the area you're targeting. and send some
thoughts and prayers...

**Reply by hasantahir** [reply]
Thoughts and prayers™️. :-)

**Reply by godbout** [reply]
just a reminder that my apps (kV, Wooshy and Scrolla) are long term apps. so my goal is for the UX to be as smooth as possible for the user. so i do (and will) contact the dev of other apps when they suck at handling the macOS Accessibility. but of course after that the ball is not in my camp. (and i consider my job done as long as they don't reply.)
